sim/src/util/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+SET(LIB_NAME util)
+SET(SOURCES typeMap.h)
+SET(TESTS typeMap.test.cpp)
+
+add_library(
+    ${LIB_NAME}
+    ${SOURCES}
+)
+
+add_library(
+    ${LIB_NAME}-test
+    ${SOURCES}
+    ${TESTS}
+)
+
+# Cmake could not determine the language used from the files
+# https://stackoverflow.com/questions/11801186/cmake-unable-to-determine-linker-language-with-c
+set_target_properties(${LIB_NAME} PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
Is there anyway we can avoid this CmakeList boilerplate for every  folder we create?

Reply:
I doubt it. I don't think CMakeLists files can export variables to the parent scope. So, the child scope has to handle all the files defined in `SOURCES` and `TESTS`.